sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -308,6 +308,99 @@ export function getCountDetailsOrUndefined(value: any): MessageCountDetails | un
   };
 }
 
+/**
+ * Service expects the time duration in ISO-8601 time duration format.
+ * Example: PT10M, P106DT2H48M5.47S
+ *
+ * This helper method is to convert the duration in seconds to ISO-8601 duration format.
+ *
+ * https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm
+ *
+ * @export
+ * @param {(number | undefined)} timeInSeconds
+ * @returns {(string | undefined)}
+ */
+export function getISO8601DurationFromSeconds(
+  timeInSeconds: number | undefined
+): string | undefined {
+  if (timeInSeconds == null) {
+    return undefined;
+  }
+  if (typeof timeInSeconds != "number") {
+    throw new Error(`Invalid input for time duration - ${timeInSeconds}`);
+  }
+  const second = { label: "S", inSeconds: 1 };
+  const minute = { label: "M", inSeconds: 60 };
+  const hour = { label: "H", inSeconds: minute.inSeconds * 60 };
+  const day = { label: "D", inSeconds: hour.inSeconds * 24 };
+  const countDecimals =
```

Comment:
I'm not really sure why we get how many decimal values there are, just to call toFixed of that value later. Are we trying to truncate it in a way that avoids floating point issues?

What's the case where this isn't true? 

```ts
timeInSeconds.toString() === timeInSeconds.toFixed(timeInSeconds.toString().split(".")[1].length)
```